Automattic/wp-calypso
# Improve Pattern Assembler Performance ⚡️

## Proposed Changes

This moves data around in a smarter way to save on network requests and speed up the site-setup flow, which includes the pattern assembler. It does three things:

### No longer sets global styles if none are set
- Before, we sent two GET and POST requests to `global-styles` even when none are set. These requests serve no purpose. Because applying a theme resets the global styles anyway.

### No longer sends a request to get global styles ID
- Before this PR, we sent a blocking request to get the GS post ID, this uses the ID value from setting the theme. It was added here: D105016-code. This one request was taking 800ms.

### No longer send a GET request /themes/mine after setting the theme
1. Before this PR, we set the theme using `setThemeOnSite` which sends a POST request to `themes/mine` then we'd send another GET request to `themes/mine` to get the theme that was applied. The POST request returns the same data as the GET request.
2. Now we use the return of the POST request instead, and got rid of the blocking GET request.

### Collapses three requests into one
These requests were being sent individually, now we use `onboard-customizations` instead. 

- Setting site intent.
- Setting site goals
- Setting launchpad_screen option

## Testing Instructions
1. Please go through the pattern assembler flow using the CFT steps here: pdtkmj-1eO-p2 but using the Calypso live link. Make sure the header, footer, homepage, and global styles are set. 
2. Please go through the pattern assembler flow but use a non-Virtual design (e.g., Appleton), the theme should be applied correctly.
